samples/subsys/lorawan building failed

**Describe the bug**
Error raised on  `west build -p -b nucleo_wl55jc samples/subsys/lorawan/class_a/

Please also mention any information which could help others to understand
the problem you're facing:
 - platform :  nucleo_wl55jc
 -  following  commit a6a4400b8621bc618c60561c5bdd90b2e8ce09e0

**To Reproduce**
Steps to reproduce the behavior:
1. west build -p -b nucleo_wl55jc samples/subsys/lorawan/class_a/
2. See error

**Impact**
Ci failure on  testcase samples/subsys/lorawan/class_a/


**Logs and console output**
```
./include/zephyr/drivers/lora.h:132:43: warning: 'struct k_poll_signal' declared inside parameter list will not be visible outside of this definition or declaration
  132 |                                    struct k_poll_signal *async);
      |                                           ^~~~~~~~~~~~~
./include/zephyr/drivers/lora.h:142:30: error: unknown type name 'k_timeout_t'
  142 |                              k_timeout_t timeout, int16_t *rssi, int8_t *snr);
      |                              ^~~~~~~~~~~
/./include/zephyr/drivers/lora.h:166:9: error: unknown type name 'lora_api_recv'
  166 |         lora_api_recv recv;
      |            ^~~~~~~~~~~~~

``` 

**Environment (please complete the following information):**
 - OS: ubuntu 
 - Toolchain Zephyr SDK 0.15.1
 - Commit SHA b219b5cba721b43a2c4fe71f5845f0abc852e7f6

--- a/include/zephyr/drivers/lora.h
+++ b/include/zephyr/drivers/lora.h
@@ -15,7 +15,8 @@
  * @{
  */
 
-#include <zephyr/types.h>
+#include <stdint.h>
+#include <zephyr/kernel.h>
 #include <zephyr/device.h>
 
 #ifdef __cplusplus

--- a/subsys/lorawan/lorawan.c
+++ b/subsys/lorawan/lorawan.c
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <zephyr/kernel.h>
 #include <zephyr/init.h>
 #include <errno.h>
 #include <zephyr/lorawan/lorawan.h>

